pages/behaviors/[behaviorId].tsx

```diff
@@ -4,47 +4,56 @@ import Navbar from "../../components/Navbar";
 import Link from "next/link";
 import { Button } from "@material-ui/core";
 import CheckUser from "../../auth0CheckUser";
+import { useState, useEffect } from "react";
+import { useRouter } from 'next/router';
 
-function BehaviorPage({ behaviorData: { data } }) {
+function BehaviorPage() {
     // Verifies if user has the correct permissions
     const {allowed, role} = CheckUser(["Admin", "BCBA"])
     if(!allowed) return(<div>Redirecting...</div>);
+
+    const router = useRouter();
+    const { behaviorId } = router.query;
+
+    const [behaviorData, setBehaviorData] = useState([])
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const response = await fetch(`/api/search/behavior?id=${behaviorId}`, { method: 'GET' });
+                if (response.ok) {
+                    const data = await response.json();
+                    setBehaviorData(data);
+                } else {
+                    console.error('Failed to fetch data:', response.status, response.statusText);
+                }
+            } catch (error) {
+                console.error('Error fetching data:', error);
+            }
+        };
+        fetchData();
+    }, [])
     
     return (
         <>
             <Head>
                 <title>Behavior</title>
                 <link rel="icon" href="/atc-logo.png" />
             </Head>
-            <Navbar pageTitle={data.name} role={role}>
+            <Navbar pageTitle={behaviorData.name} role={role}>

```

Comment:
Optional, but you might want to consider adding a loading fallback. If the `behaviorData` is null, display something like `Loading...` or if you feel fancy you can also put a loading spinner. If you feel even fancier, look at this: https://mui.com/material-ui/react-skeleton/